### PR TITLE
NPM Mirroring

### PR DESCRIPTION
**Pull request by lholmquist**

### CODE REVIEW

**Comment by domharries** [comment]
A couple of links to the Garage pages on Nexus/Artifactory. They don't have a huge amount of detail so may not be worth including:

- https://www.ibm.com/garage/method/practices/deliver/tool_nexus/
- https://www.ibm.com/garage/method/practices/deliver/tool_artifactory/

File: docs/development/npm-mirroring.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# NPM Mirroring
```
Comment (lholmquist) [inline docs/development/npm-mirroring.md:1]: Update to internal registry/proxy

**Comment by lholmquist** [comment]
Add somehting related to rate limits on npm @lholmquist

**Comment by lholmquist** [comment]
@lholmquist commercial vs. opensource versions of artifactory/nexus

File: docs/development/npm-proxy.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+* Using a proxy/mirror can provide a centralized point for scanning for security vulnerabilities
+
+* A mirror can reduce the dependecy on the public registry.
+
+* You need to maintain a copy of a module incase it is removed from the public registry.
+
+* Being a good npm citizen.  The public registry is a free service and npm allows for [update to 5 million requests per month](https://blog.npmjs.org/post/187698412060/acceptible-use), which can be used up quickly with CI builds.
+
+Using a npm mirror/proxy is fairly easy.  You can set the *registry* that the npm cli uses by running `npm set registry URL`.  You can also use the `HTTP_PROXY` and `HTTPS_PROXY` environment variables to set the mirrored registry.
```
Comment (lholmquist) [inline docs/development/npm-proxy.md:27]: remove sentence about HTTP_Proxy

File: docs/development/npm-proxy.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## Learning Resources
+
+https://jfrog.com/artifactory/
+
+https://guides.sonatype.com/repo3/quick-start-guides/proxying-maven-and-npm/
+
+https://www.sonatype.com/products/repository-oss-vs-pro-features
+
+https://verdaccio.org/
```
Comment (lholmquist) [inline docs/development/npm-proxy.md:39]: remove https://verdaccio.org/

**Review by wtrocki — APPROVED** [review]

File: docs/development/npm-proxy.md (hunk 3 above)
Comment (mhdawson) [inline docs/development/npm-proxy.md:39]:
```suggestion
https://verdaccio.org/
```

File: docs/development/npm-proxy.md (hunk 2 above)
Comment (mhdawson) [inline docs/development/npm-proxy.md:27]:
```suggestion
Using a npm mirror/proxy is fairly easy.  You can set the *registry* that the npm cli uses by running `npm set registry URL`. 
```

**Review by mhdawson — APPROVED** [review]
LGTM once suggestions are incorporated.

**Pull request merged**